### DISCUSSION TITLE
OpenTelemetry AWS X-Ray Trace ID Support

### DISCUSSION THREAD

**Discussion by lewis262626** [grafana/grafana#60884] [Platform]
Using the `tracing.opentelemetry.otlp.propogator` setting Grafana supports tracing using the `w3c` and `jaeger` traceID format and propagation. AWS have defined [X-Ray](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html) traceIDs as 128 bits, but the first 32 bits are the unix epoch time of when the trace was created. Since X-Ray only allows submission of Trace IDs from the past 30 days, received Trace IDs are checked. 

Could support be added for X-Ray ID generation and propogation using the `X-Amzn-Trace-Id` HTTP header.

There's already a [Go library for this, so should be pretty simple to implement](https://pkg.go.dev/go.opentelemetry.io/contrib/propagators/aws/xray)

[More documentation on the ADOT Go documentation](https://aws-otel.github.io/docs/getting-started/go-sdk/trace-manual-instr)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82820. Feel free to continue the discussion around this there. Thank you!
